docs/Developers/Git_Basics.rst

```diff
@@ -70,7 +70,9 @@ Creating a new branch
 ---------------------
 
 To start working on an issue, you first need to create a new branch where you
-will work.
+will work. Why make a new branch? If you develop on the master branch, you can
+and will end up having issues with your commits. This is especially prevalent
+if you are changing multiple features.
```

Comment:
Elaborate why having issues in the master branch is bad. Something like "You should always have one clean and working branch which is considered to be master". Just word it better :D